crates/youki/src/commands/spec_json.rs

```diff
@@ -106,7 +110,9 @@ mod tests {
         let spec = get_rootless()?;
         let tmpdir = create_temp_dir("test_spec_json").expect("failed to create temp dir");
         let path = tmpdir.path().join("config.json");
-        to_writer_pretty(&File::create(path)?, &spec)?;
+        let file = File::create(path)?;
+        let writer = BufWriter::new(file);
+        to_writer_pretty(writer, &spec)?;
```

Comment:
Hey, I know this is a test, but should we not also add a `writer.flush()` at the end, like we have done at other places?

Reply:
https://doc.rust-lang.org/nightly/std/io/struct.BufWriter.html
It is critical to call [flush](https://doc.rust-lang.org/nightly/std/io/struct.BufWriter.html#method.flush) before BufWriter<W> is dropped. Though dropping will attempt to [flush](https://doc.rust-lang.org/nightly/std/io/struct.BufWriter.html#method.flush) the contents of the buffer, any errors that happen in the process of dropping will be ignored. Calling flush ensures that the buffer is empty and thus dropping will not even attempt file operations.

Reply:
Yes, so we should add a flush call after `to_writer_pretty` like other places right?

Reply:
Yes.

Reply:
So can you update and push it, so we can go ahead and merge this?

Reply:
Updated.